corehq/apps/importer/tasks.py

```diff
@@ -82,12 +83,20 @@ def _submit_caseblocks(domain, case_type, caseblocks):
                 )
             else:
                 properties = set().union(*map(lambda c: set(c.dynamic_case_properties().keys()), cases))
-                add_inferred_export_properties.delay(
-                    'CaseImporter',
-                    domain,
-                    case_type,
-                    properties,
-                )
+                if case_type and len(properties):
+                    add_inferred_export_properties.delay(
+                        'CaseImporter',
+                        domain,
+                        case_type,
+                        properties,
+                    )
+                elif len(properties):
+                    _soft_assert = soft_assert(notify_admins=True)
+                    soft_assert(
+                        False,
```

Comment:
it would be a little cleaner if you actually replaced `False` with `len(properties) == 0` which is the assertion we are hoping to check, and then we can get rid of the `elif` and use `soft_assert` more like an assertion